compiler/rustc_hir_analysis/src/check/region.rs

```diff
@@ -887,3 +897,22 @@ pub fn region_scope_tree(tcx: TyCtxt<'_>, def_id: DefId) -> &ScopeTree {
 
     tcx.arena.alloc(scope_tree)
 }
+
+/// Check for use of mutable static
+pub fn static_mut_ref(tcx: TyCtxt<'_>, stmts: &[Stmt<'_>]) {
+    for stmt in stmts {
+        if let hir::StmtKind::Local(loc) = stmt.kind
+            && let Some(init) = loc.init
+            && let hir::ExprKind::AddrOf(borrow_kind, _, expr) = init.kind
+            && matches!(borrow_kind, BorrowKind::Ref)
+            && let hir::ExprKind::Path(qpath) = expr.kind
+            && let hir::QPath::Resolved(_, path) = qpath
+            && let hir::def::Res::Def(def_kind, _) = path.res
+            && let hir::def::DefKind::Static(mt) = def_kind
+            && matches!(mt, Mutability::Mut)
+        {
+            let span = init.span;
+            tcx.sess.emit_err(errors::StaticMutRef { span });
```

Comment:
could you move the edition check here, and then either emit a lint or emit the error depending on whether the edition is 2024 or above?